### PR TITLE
ci(pre-push hook): automate loading pre-push hook with cargo-husky

### PR DESCRIPTION
**Pull request by SLASHLogin**
Add cargo-husky to dev-deps, create hook, update `CONTRIBUTING.md`

#214

### CODE REVIEW

**Review by mindoodoo — APPROVED** [review]
Awesome stuff, thanks !

**Comment by SLASHLogin** [comment]
Please, do not merge this just yet. I think rusty-hooks might be a better choice for this task, as the developer behind it seems to be more known in the rust community. Even though both crates didn't receive much attention recently, I think there's just a bigger possibility that `rusty-hook`'s maintainer would at least respond. I'd like to test how it behaves on my branch with CI first, tho.

File: Cargo.toml
Comment (SLASHLogin) [reply Cargo.toml]: Removed in last commit

**Review by orhun — APPROVED** [review]

File: Cargo.toml
Comment (joshka) [inline Cargo.toml:60]:
```suggestion

cargo-husky = { version = "1.5.0", default-features = false, features = ["user-hooks"] }
```

**Comment by joshka** [comment]
I squashed the commits and added the following changes to this:
- used the one line suggestion in the cargo toml
- changed this so it prints a message rather than installing cargo-make as we shouldn't generally force installation of tools on people's machines.

**Review by joshka — APPROVED** [review]

**Review by orhun — APPROVED** [review]

**Pull request merged**
